### PR TITLE
 Scrying orb now grants sixth sense (and xray) to whoever possesses it

### PR DESCRIPTION
**Pull request by coiax**
:cl:
add: A wizard's scrying orb now grants both xray vision, and the ability to hear the dead to whoever possesses it.
balance: However, as soon as the orb is no longer in your possession, these abilities fade.
/:cl:

This was @XDTM's idea, when he implemented near death experiences.

It works even when in your backpack.

This also adds the TRAIT_XRAY_VISION, for non-genetic, non-eyeball sources of xray.

Why? Having a wizard talking to ghosts means it looks like he's talking to thin air. Also, this makes scrying orb a little more active in its knowledge gathering. Ghosts will have the chance to mislead the wizard with whatever they're talking about.

### CODE REVIEW

**Comment by zxaber** [comment]
Does the orb allow hearing all dedchat, or just the ghosts nearby? Would localized ghost hearing be a neat change, or would it just make using the orb tedious?

Also, does this replace the old function of temporarily ghosting?

**Comment by coiax** [comment]
Old function of temporarily ghosting is still there.

If the wizard didn't want to be haunted by the voices of the dead, he shouldn't have bought it.

**Comment by Ratvar** [comment]
Well, before you could just *not ghost*.

**Comment by zxaber** [comment]
You could also probably just hide it somewhere. Stick it in a toilet?

**Comment by Ratvar** [comment]
Buff at cost of chat spam is... why not add a toggle to ghost hearing/make so its inhand only, again?

**Comment by ishitbyabullet** [comment]
Thought you already got x-ray. I had a scrying orb once and it gave me x-ray vision. It went away randomly, though.

**Comment by coiax** [comment]
The current behaviour is the first human to pick up the scrying orb gets the xray mutation. But that'll go away with exposure to mutadone, cloning, that sort of thing.

With these changes, the xray vision is no longer genetic, but only lasts while you have the orb.

**Comment by ExcessiveUseOfCobblestone** [comment]
@ishitbyabullet whoever gets it first used to get xray that persisted even if you lost the orb, now it's whoever has it at all but you lose it upon losing the orb.

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -590,12 +590,18 @@
...
 		if(!isnull(G.lighting_alpha))
 			lighting_alpha = min(lighting_alpha, G.lighting_alpha)
-	if(dna)
+
+	var/has_xray = has_trait(TRAIT_XRAY_VISION)
+
+	if(dna && !has_xray)
 		for(var/X in dna.mutations)
 			var/datum/mutation/M = X
 			if(M.name == XRAY)
```
Comment (XDTM) [inline code/modules/mob/living/carbon/carbon.dm]: might be a good moment to make the mutation give the trait directly, like hulk does

**Pull request merged**